src/main/java/by/exadel/internship/service/InternshipService.java

```diff
@@ -37,16 +42,66 @@ public GuestInternshipDTO getById(UUID uuid) {
     }
 
     public List<GuestInternshipDTO> getAll() {
-
+        putClassNameInMDC();
         MDC.put("className", InternshipService.class.getSimpleName());
         log.info("Try to get all Internships");
-
         List<Internship> internships = internshipRepository.findAll();
         List<GuestInternshipDTO> guestInternshipDTOList = internShipMapper.map(internships);
-
+        log.info("Successfully got list of Internships");
         log.info("Successfully list of Internships");
-
         return guestInternshipDTOList;
 
     }
+
+    public List<GuestInternshipDTO> getAllDeleted() {
+        putClassNameInMDC();
+        log.info("Try to get deleted Internships");
+        List<Internship> deletedInternships = internshipRepository.findAllDeleted();
+        List<GuestInternshipDTO> guestDeletedInternshipDTOList = internShipMapper.map(deletedInternships);
+        log.info("Successfully got list of deleted Internships");
+        return guestDeletedInternshipDTOList;
+    }
+
+    public GuestInternshipDTO getDeletedInternshipById(UUID uuid) {
+        putClassNameInMDC();
+        log.info("Try to get deleted Internships with uuid= " + uuid);
+        Internship deletedInternship = internshipRepository
+                .findDeletedById(uuid)
+                .orElseThrow(() -> new NotFoundException("No such deleted Internship with uuid = " + uuid +
+                        " in DB", "uuid.invalid"));
+        log.info("Successfully got deleted Internships with uuid= " + uuid);
+        GuestInternshipDTO guestDeletedInternshipDTO = internShipMapper.toGuestInternshipDTO(deletedInternship);
+        return guestDeletedInternshipDTO;
+    }
+
+    @Transactional
+    public GuestInternshipDTO restoreInternshipById(UUID uuid) {
+        putClassNameInMDC();
+        log.info("Try to return deleted Internship with uuid= " + uuid + " to List if Internships");
+        Internship internship = internshipRepository
+                .findDeletedById(uuid)
+                .orElseThrow(() -> new NotFoundException("No such deleted Internship with uuid = " + uuid +
+                        " in DB", "uuid.invalid"));
+        internshipRepository.updateDeletedById(uuid);
+        internship.setDeleted(false);
+        log.info("Successfully returned deleted Internship with uuid= " + uuid);
+        return internShipMapper.toGuestInternshipDTO(internship);
+
+    }
+
+    public void deleteInternshipById(UUID internshipId) {
+        putClassNameInMDC();
+        log.info("Try to delete Internship with uuid= " + internshipId);
+        try {
+            internshipRepository.deleteById(internshipId);
+        }catch (EmptyResultDataAccessException exception){
```

Comment:
It's good first of all to find entity by id, and after then it try to remove it.
When you provide `catch (EmptyResultDataAccessException exception)` you don't pay attention that you are working through the spring data interface and different databases can throw different errors, hibernate developers can add / change error naming/designation ect. and after such modifications you can get bug.

Reply:
I didn't think about it. Fixed